--- vendor/phpunit/phpunit/src/Util/Configuration.php	2018-02-01 06:50:59.000000000 +0100
+++ vendor/phpunit/phpunit/src/Util/Configuration.php	2021-09-17 13:28:04.000000000 +0200
@@ -159,7 +159,7 @@
         $this->xpath    = new DOMXPath($this->document);
     }
 
-    final private function __clone()
+    private function __clone()
     {
     }
 
@@ -501,7 +501,7 @@
             // See https://github.com/sebastianbergmann/phpunit/issues/277
             switch ($array) {
                 case 'var':
-                    $target = &$GLOBALS;
+                    //$target = &$GLOBALS;
                     break;
 
                 case 'server':
